Hmac.js

```diff
@@ -87,7 +84,7 @@ export class Hmac {
       }
       return signature;
     } finally {
-      this._requests.delete(cacheKey);
+      this.cache.delete(requestCacheKey);
```

Comment:
Let's just have one cache key ... and it returns a promise. We don't need this extra thing, right?